emcc.py

```diff
@@ -1674,6 +1674,9 @@ def parse_passes(string):
           if shared.Settings.SAFE_HEAP:
             passes += ['--safe-heap']
           passes += ['--post-emscripten']
+          # always inline __original_main into main, as otherwise it makes debugging confusing,
+          # and doing so is never bad for code size
+          passes += ['--inline-main']
```

Comment:
By having it be a generic pass, we let non-emscripten compilers use it easily as well.

Reply:
Sure, but woudn't it make sense to have all the generatic passes the emscripten relies on autoamically run by wasm-emscripten-finalize?   To put it another way:  why not have wasm-emscripten-finalize run this pass?

Reply:
Oh, I see. Well, maybe - we could move all the passes from this section of emcc.py into wasm-emscripten-finalize, I suppose. That's a larger design question, I guess, of whether we want to drive binaryen passes from emscripten or binaryen. I lean towards the current location, as some of the passes require information about emscripten flags - if we moved it all to wasm-emscripten-finalize, we'd need to pipe all those options over to there.

Reply:
My understanding is that `'--post-emscripten` is kind of the exception because there was some reason why it can't just go in `wasm-emscripten-finalize`.   All the other passes just get run unconditionally in `wasm-emscripten-finalize`. 

I for one think that if we can reduce the linking process to basically "lld + finalize" that is way better than running lot of different opt / binaryen runs.

Reply:
I think we could move post-emscripten to finalize actually. At least I can't think of offhand why not.

But many other passes in the surrounding code depend on emscripten flags, like

 * `--no-exit-runtime` depends on `NO_EXIT_RUNTIME`
 * `--safe-heap` depends on `SAFE_HEAP`
 * `--strip-debug` depends on the debug level
 * `--strip-producers` depends on `EMIT_PRODUCERS_SECTION`

etc. And the order of those passes is interconnected, so it's hard to do half on one side and half on the other. Seems most straightforward to drive it all from the place with all the information already.

Reply:
@sbc100 in any case, changing how we drive binaryen passes from emcc seems like a larger discussion than this PR?